src/server.cpp

```diff
@@ -1327,13 +1328,7 @@ void Server::setInventoryModified(const InventoryLocation &loc, bool playerSend)
 		break;
 	case InventoryLocation::NODEMETA:
 	{
-		v3s16 blockpos = getNodeBlockPos(loc.p);
-
-		MapBlock *block = m_env->getMap().getBlockNoCreateNoEx(blockpos);
-		if(block)
-			block->raiseModified(MOD_STATE_WRITE_NEEDED);
```

Comment:
same here, we have to take care of storage of the modified block in the map database. We can spare sending it, thats fine, but we shouldn't avoid storing them into the map.